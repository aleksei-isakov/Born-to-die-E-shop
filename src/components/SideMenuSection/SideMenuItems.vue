<template>
  <div>
    <div
      v-for="(item, index) in items"
      :key="index"
      class="side-menu-item"
      @click="onClickClosePopup"
    >
      <base-button-router :path="item.path" class="side-menu-item__title">
        <i class="side-menu-item__icon" :class="`${item.icon}`"></i>
        {{ item.title }}
      </base-button-router>
    </div>
  </div>
</template>

<script>
import BaseButtonRouter from '@/base_components/BaseButtonRouter/BaseButtonRouter.vue';

export default {
  name: 'SideMenuItems',

  components: { BaseButtonRouter },

  data() {
    return {
      items: [
        {
          title: 'My profile',
          icon: 'fas fa-house-user',
          path: '/profile'
        },

        { title: 'My wishlist', icon: 'fas fa-heart', path: '/wishlist' },

        {
          title: 'My orders',
          icon: 'fas fa-folder',
          path: '/orders'
        },

        {
          title: 'My address book',
          icon: 'fas fa-address-book',
          path: '/address-book'
        }
      ]
    };
  },

  methods: {
    onClickClosePopup() {
      this.$emit('on-click-close-popup');
    }
  }
};
</script>

<style lang="scss">
@import '@/scss/CustomVariables.scss';

.side-menu-item {
  display: flex;
  padding: 9px 12px;

  &__icon {
    font-size: 20px;
    color: $primary;
    margin-right: 5px;
  }

  &__title {
    font-size: 16px;
    width: 100%;
    text-align: left;
    color: $primary !important;
  }
}
</style>
