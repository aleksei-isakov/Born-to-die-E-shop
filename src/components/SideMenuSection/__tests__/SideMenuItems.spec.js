import { shallowMount, createLocalVue } from '@vue/test-utils';
import SideMenuItems from '@/components/SideMenuSection/SideMenuItems';

describe('SideMenuItems.vue', () => {
  let wrapper;

  beforeEach(() => {
    const localVue = createLocalVue();

    wrapper = shallowMount(SideMenuItems, {
      localVue
    });
  });

  it('should render items in side menu correct', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
