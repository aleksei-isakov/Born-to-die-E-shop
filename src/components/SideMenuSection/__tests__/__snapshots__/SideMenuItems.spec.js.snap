// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenuItems.vue should render items in side menu correct 1`] = `
<div
  class="items"
>
  <div
    class="side-menu-item"
  >
    <base-button-router-stub
      class="side-menu-item__title side-menu-item__title_blue"
      path="/profile"
    >
      <i
        class="fas fa-house-user side-menu-item__icon side-menu-item__icon_blue"
      />
      
      My profile
    
    </base-button-router-stub>
  </div>
  <div
    class="side-menu-item"
  >
    <base-button-router-stub
      class="side-menu-item__title side-menu-item__title_blue"
      path="/wishlist"
    >
      <i
        class="fas fa-heart side-menu-item__icon side-menu-item__icon_blue"
      />
      
      My wishlist
    
    </base-button-router-stub>
  </div>
  <div
    class="side-menu-item"
  >
    <base-button-router-stub
      class="side-menu-item__title side-menu-item__title_blue"
      path="/orders"
    >
      <i
        class="fas fa-folder side-menu-item__icon side-menu-item__icon_blue"
      />
      
      My orders
    
    </base-button-router-stub>
  </div>
  <div
    class="side-menu-item"
  >
    <base-button-router-stub
      class="side-menu-item__title side-menu-item__title_blue"
      path="/address-book"
    >
      <i
        class="fas fa-address-book side-menu-item__icon side-menu-item__icon_blue"
      />
      
      My address book
    
    </base-button-router-stub>
  </div>
</div>
`;
